SilMock/Google/Service/Directory/Resource/GroupsAliases.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace SilMock\Google\Service\Directory\Resource;
+
+use Exception;
+use Google\Service\Directory\Alias as GoogleDirectory_GroupsAlias;
+use Google\Service\Directory\Aliases as GoogleDirectory_GroupsAliases;
+use SilMock\Google\Service\DbClass;
+use SilMock\Google\Service\Directory\ObjectUtils;
+
+class GroupsAliases extends DbClass
+{
+    public function __construct(?string $dbFile = null)
+    {
+        parent::__construct($dbFile, 'directory', 'groupsaliases');
+    }
+
+    public function delete(string $groupKey, string $alias, array $optParams = [])
+    {
+        $groupRecords = $this->getRecords();
+        foreach ($groupRecords as $groupRecord) {
+            $groupRecordData = json_decode($groupRecord['data'], true);
+            if (
+                $groupRecordData['primaryEmail'] === $groupKey
+                && $groupRecordData['alias'] === $alias
+            ) {
+                $this->deleteRecordById($groupRecord['id']);
+            }
+        }
+    }
+
+    /**
+     * @throws Exception
+     */
+    public function insert(
+        string $groupKey,
+        GoogleDirectory_GroupsAlias $postBody,
+        array $optParams = []
+    ): GoogleDirectory_GroupsAlias {
+        if ($this->isNewGroupAlias($groupKey, $postBody->getAlias())) {
+            $postBody['id'] = $postBody['id'] ?? microtime();
```

Comment:
How likely is it to batch add new group aliases? Would microtime be able to distinguish between them (that is, could it take less than a microsecond to create a new alias and attempt to make a new group alias with the same id)?

Reply:
MailAdmin will only be adding one alias. So, I don't expect this to be an issue.